File: components/delete.js
import { delete as del } from 'sql-bricks';
import component from './lib/table';
import { parseSQLError } from './lib/parse';
import getConnection from './lib/getConnection';

export default async (e, context, callback) => {
  const { id } = e.pathParameters;
  const sql = del(component.name).where({ id });
  const conn = await getConnection();

  let response;
  const headers = { 'Access-Control-Allow-Origin': '*' };

  try {
    await conn.query(sql.toString());

    response = {
      body: JSON.stringify({ data: true }),
      headers,
      statusCode: 204,
    };
  } catch (error) {
    response = {
      body: JSON.stringify({ error: parseSQLError(error) }),
      headers,
      statusCode: 400,
    };
  } finally {
    conn.end();
  }

  callback(null, response);
};

File: works/delete.py
import sys
import jwt
import json
from datetime import datetime
from contrib import db, abort, new_error, jwt_decode, can_delete


def handler(event, context):
    headers = event.get('headers') or {}
    user_id = jwt_decode(headers.get('Authorization')).get('user_id')
    path_params = event.get('pathParameters') or {}
    work_id = path_params.get('id')

    work = db.table('work').where({'id': work_id}).first()

    if work is None:
        return abort(404, new_error('invalid id', 1))

    if not can_delete(work_id, user_id):
        return abort(404, new_error('no permissions', 2))

    db.table('work').where({'id': work_id}).update({
        'deletedAt': datetime.now(),
    })

    return {
        'body': json.dumps({'data': True}),
        'headers': {'Access-Control-Allow-Origin': '*'},
        'statusCode': 200,
    }
